src/main/java/com/github/tcdl/jsonmask/JsonMask.java

```diff
@@ -0,0 +1,50 @@
+package com.github.tcdl.jsonmask;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class JsonMask {
+
+    public static JsonNode mask(JsonNode target) {
+        return traverseAndMask(target.deepCopy());
+    }
+
+    private static JsonNode traverseAndMask(JsonNode target) {
+        if (target.isTextual()) {
+            return new TextNode(maskString(target.asText()));
+        }
+        if (target.isNumber()) {
+            return new TextNode(maskNumber(target.asText()));
+        }
+
+        if (target.isObject()) {
+            Iterator<Map.Entry<String, JsonNode>> fields = target.fields();
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> field = fields.next();
+                ((ObjectNode) target).replace(field.getKey(), traverseAndMask(field.getValue()));
+            }
+        }
+        if (target.isArray()) {
+            for (int i = 0; i < target.size(); i++) {
+                ((ArrayNode) target).set(i, traverseAndMask(target.get(i)));
+            }
+        }
+        return target;
+    }
+
+    private static String maskString(String value) {
+        return value
+                .replaceAll("\\d", "*")
+                .replaceAll("[A-Z]", "X")
+                .replaceAll("[^X\\s!-/:-@\\[-`{-~]", "x");
```

Comment:
As a little optimization, we could pre-compile regexes (i.e. use `Pattern.compile()`)